wcfsetup/install/files/lib/action/AJAXProxyAction.class.php

```diff
@@ -114,9 +115,14 @@ protected function sendResponse()
             }
         }
 
-        // force background queue invocation
-        if (!\class_exists(WCFACP::class, false) && WCF::getSession()->getVar('forceBackgroundQueuePerform')) {
-            $this->response['forceBackgroundQueuePerform'] = true;
+        if (!RequestHandler::getInstance()->isACPRequest() && WCF::getSession()->getVar('forceBackgroundQueuePerform')) {
```

Comment:
As this is modified anyway: This could also check `BackgroundQueueHandler::getInstance()->hasPendingCheck()`.

Reply:
I have considered this, but opted out. First of all, this is as close to the existing implementation and second I consider the session variable to be a superset of `hasPendingCheck()`.

At worst we’re invoking the background queue “twice” in case there is a concurrent request that enabled that flag. And there is no such thing as “invoking the background queue _too often_”.

Reply:
Okay, fair enough.